torch/lib/c10d/Utils.hpp

```diff
@@ -31,6 +31,39 @@ typedef SSIZE_T ssize_t;
 
 namespace c10d {
 
+// Distributed c10d debug levels
+enum DistributedDebugLevel {
+  OFF = 0,
+  DETAIL = 1,
+  INFO = 2,
+};
+
+// String debug log levels
+extern const char * kDistDebugEnvVar;
+extern const char* kDistDebugDetailLogLevel;
+extern const char* kDistDebugInfoLogLevel;
+extern const char* kDistDebugOffLogLevel;
+
+inline DistributedDebugLevel getDebugLevel(const char * levelStr) {
+
+  static std::unordered_map<std::string, DistributedDebugLevel> mapping = {
+    {kDistDebugOffLogLevel, DistributedDebugLevel::OFF},
+    {kDistDebugInfoLogLevel, DistributedDebugLevel::INFO},
+    {kDistDebugDetailLogLevel, DistributedDebugLevel::DETAIL},\
+  };
+
+  auto it = mapping.find(levelStr);
+  TORCH_CHECK(
+    it != mapping.end(),
+    "Invalid string value for distributed debug mode: ", levelStr
+  );
```

Comment:
nit: feel this check can be merged with the check in parseDistDebugLevel(), also this function can be merged with parseDistDebugLevel() as well